### ISSUE TITLE
VictoriaLogs grouping issue

### ISSUE THREAD

**Issue by darthvader2** [VictoriaMetrics/VictoriaMetrics#7620]
### Describe the bug

In Victoria logs,  grouping element has an issue with case sensitive. I get logs from different services, some have "Hostname" and some "hostname".  So, all my logs with hostname are not getting grouped as intended.

### To Reproduce

Try to send rsyslogs to vlogs with different cases of hostname

### Version

victoria-logs-v1.0.0

**Comment by tiny-pangolin** [comment]
as a workaround would it be possible to forward to syslog to a collector like [vector](vector.dev) and use the [downcase](https://vector.dev/docs/reference/vrl/examples/#downcase) function or use the [telegraf strings processor](https://github.com/influxdata/telegraf/tree/master/plugins/processors/strings) to resolve this issue?

I'm not sure if we want to add log processing to victorialogs since there are already multiple log processing tools on the market we integrate well with.

**Comment by darthvader2** [comment]
Thank you for the suggesting vector, I do not suggest log processing to victorialogs either. My issue was, if there is an attribute , here hostname. If I want to group them it is not being shown in the dropdown menu, so  I thouht it is a kind of bug

**Comment by valyala** [comment]
VictoriaLogs doesn't provide abilities to change the case of log fields during querying. I think it would be great adding such functionality into [`format` pipe](https://docs.victoriametrics.com/victorialogs/logsql/#format-pipe) by prefixing the formatted log field with string transformation commands such as `lc` or `uc`. For example, the following [LogsQL](https://docs.victoriametrics.com/victorialogs/logsql/) query could be used for lowercasing all the `host` field values before calculating per-host number of logs over the last hour:

```logsql
_time:1h
  | format "<lc:host>" as host_lowercase
  | stats by (host_lowercase) count()
```

**Comment by valyala** [comment]
FYI, the commit 77430b797d53140d29ff9abaaee205ae7665aca4 adds the ability to convert log fields into lowercase and uppercase via `<lc:field_name>` and `<uc:field_name>` options inside [`format` pipe](https://docs.victoriametrics.com/victorialogs/logsql/#format-pipe). This commit will be included in the next release of VictoriaLogs.

This functionality can be verified before the next release by manually building VictoriaLogs from the commit mentioned above according to [these docs](https://docs.victoriametrics.com/victorialogs/quickstart/#building-from-source-code).

**Comment by valyala** [comment]
The `lc` and `uc` modifiers in the [format pipe](https://docs.victoriametrics.com/victorialogs/logsql/#format-pipe) are available in VictoriaLogs starting from [v1.1.0-victorialogs](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.1.0-victorialogs). The original issue can be resolved by the query from [this comment](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7620#issuecomment-2502170924).
Closing the issue as resolved.
